hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3RepositoryAutoConfiguration.java

```diff
@@ -77,11 +81,16 @@ public ClientConfiguration awsClientConfiguration() {
     @Bean
     @ConditionalOnMissingBean
     public AmazonS3 amazonS3() {
-        AmazonS3ClientBuilder s3ClientBuilder = AmazonS3ClientBuilder.standard()//
-                .withCredentials(awsCredentialsProvider())//
+        AmazonS3ClientBuilder s3ClientBuilder = AmazonS3ClientBuilder.standard()
+                .withCredentials(awsCredentialsProvider())
                 .withClientConfiguration(awsClientConfiguration());
         if (!StringUtils.isEmpty(region)) {
             s3ClientBuilder = s3ClientBuilder.withRegion(region);
+            if (!StringUtils.isEmpty(endpoint)) {
+                s3ClientBuilder = s3ClientBuilder
+                    .withPathStyleAccessEnabled(true)
```

Comment:
I do understand that path style access is very convenient in combination with S3-like storages. However S3 itself deprecated the use of `path style` access in favour of `virtual-hosted` style (cf. [AWS path deprecation](https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story)).

It seems that MinIO supports `virtual-hosted` style as well (cf. [MinIO](https://github.com/minio/minio/tree/master/docs/config#domain)):

> MINIO_DOMAIN environment variable is used to enable virtual-host-style requests

I assume using this approach will be more future proof (especially in terms of using the AWS SDK for other storage options). Furthermore it would allow the hawkBit S3 extension to stay flexible enough for supporting both access styles in case somebody sets the endpoint property

Reply:
So I just need to remove `.withPathStyleAccessEnabled(true)` ?

Reply:
Yes, would be great if this is feasible for you. But I can't tell if it will work out of the box with MinIO. At least the AWS SDK states:
>The default behaviour is to detect which access style to use based on the configured
endpoint (an IP will result in path-style access) and the bucket being accessed

So I would assume that the SDK will choose the correct style automatically. Besides that nothing to complain :)